### PR TITLE
Husky ur5 addition & changes

### PR DESCRIPTION
**Pull request by UltronDestroyer**

### CODE REVIEW

File: urdf/husky_ur5/base_with_manipulator.urdf.xacro [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0"?>
```
Comment (paulbovbel) [inline urdf/husky_ur5/base_with_manipulator.urdf.xacro]: Instead, modify the urdf/description.xacro, something along the lines of

 <xacro:arg name="ur5_enabled" default="false" />
 <xacro:husky/>
 <xacro:if value="$(arg ur5_enabled)">
 <xacro:include ur stuff/>
 <xacro:ur5_robot/>
 <xacro:include filename="...husky_ur5_glue.xacro">

File: launch/husky_ur5_upload.launch [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
```
Comment (paulbovbel) [inline launch/husky_ur5_upload.launch]: Instead, add onto description.launch with an arg for ur5_enabled, that gets passed to description xacro. That way we have one top-level description.launch

File: urdf/husky_ur5/ur5.urdf.xacro [hunk 3]
```diff
@@ -0,0 +1,319 @@
+<?xml version="1.0"?>
```
Comment (paulbovbel) [inline urdf/husky_ur5/ur5.urdf.xacro]: File shouldn't be here

Comment (paulbovbel) [inline urdf/husky_ur5/ur5.urdf.xacro]: This file shouldn't be here

Comment (UltronDestroyer) [inline urdf/husky_ur5/ur5.urdf.xacro]: Fixed in f8327ffada3fdf415f0953acf7e11b1f5b8f27a6

File: urdf/husky_ur5/ur5_arm.urdf.xacro [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  <!-- Join arm -->
+  <joint name="husky_ur5_arm_joint" type="fixed">
+    <parent link="top_plate_link" />
+    <child link="ur5_arm_base_link" />
+    <origin xyz="0.3 ${reflect*0.005} 0.25" rpy="0.0 0.0 ${-pi/2}" />
+  </joint>
+
+  <!-- Gazebo references for self collide -->
+  <gazebo reference="ur5_arm_shoulder_link">
```
Comment (paulbovbel) [inline urdf/husky_ur5/ur5_arm.urdf.xacro]: Anything gazebo related should really be in husky_gazebo.

File: urdf/husky_ur5/ur5_arm.urdf.xacro [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+<robot xmlns:xacro="http://ros.org/wiki/xacro" name="husky_ur5">
+  <xacro:property name="reflect" value="-1" />
+
+ <xacro:include filename="$(find ur_description)/urdf/gazebo.urdf.xacro" /> 
+  <xacro:include filename="$(find ur_description)/urdf/ur5.urdf.xacro" />
+
+  <!-- Spawn arm -->
+  <xacro:ur5_robot prefix="ur5_arm_" joint_limited="true"/>
+  <!-- <xacro:ur10_robot prefix="" joint_limited="true" /> -->
```
Comment (paulbovbel) [inline urdf/husky_ur5/ur5_arm.urdf.xacro]: Please remove anything commented/unused

File: urdf/ur5_arm.urdf.xacro [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+This program contains the description of the husky_ur5 manipulator.
+-->
+
+<robot xmlns:xacro="http://ros.org/wiki/xacro" name="husky_ur5">
+  <xacro:property name="reflect" value="-1" />
+
+ <xacro:include filename="$(find ur_description)/urdf/gazebo.urdf.xacro" />
```
Comment (paulbovbel) [inline urdf/ur5_arm.urdf.xacro]: gazebo stuff into husky_simulator

Comment (UltronDestroyer) [inline urdf/ur5_arm.urdf.xacro]: Fixed

**Comment by paulbovbel** [comment]
Still missing the kinect. You need some kind of sensor to run the demos right?

**Comment by paulbovbel** [comment]
Can you add the top plate as a standard piece of the base description?

**Comment by UltronDestroyer** [comment]
@paulbovbel 

Ilia will be using only the arm for the demo, so no sensor arrays/config are required at this point in time. The top plate has been added to the husky_description as "ur5_top_plate.stl" https://github.com/husky/husky_description/blob/husky_ur5/meshes/ur5_top_plate.stl

**Comment by paulbovbel** [comment]
Fixed up, major outstanding TODOs are to find a better place for the laser, and use box models for collisions instead of STLs. Complex collision model will really bog down gazebo even more.

See https://github.com/husky/husky_description/blob/indigo-devel/urdf/husky.urdf.xacro#L71

**Comment by paulbovbel** [comment]
Still a few TODOs left regarding collision boxes.

**Comment by UltronDestroyer** [comment]
Ok, it is now a box.

**Comment by UltronDestroyer** [comment]
Alright kinect has optenv, i think that marks all of them.

**Comment by UltronDestroyer** [comment]
Alright gripper code uncommented.. just going to put the updated stow in there.

**Comment by UltronDestroyer** [comment]
Stow script has been added. You can change the values to how you want the arm to be stowed, but it no longer sits out infront.

I've to add it to the launch file yet.

**Comment by UltronDestroyer** [comment]
This can be merged.

**Comment by paulbovbel** [comment]
Please remove all the gripper stuff. When you're ready to add the gripper, make another branch and another pull. If we're gonna release this stuff, it should be in a finished state.

The kinect optenv is on the visual tag, in the macro. It should really be here https://github.com/husky/husky_description/pull/11/files#diff-ba69af255156f3b481175d26f1dace9bR176 to be effective. Otherwise you're poisoning the macro with husky-specific stuff, ideally these should be merged out into a 'clearpath sensor descriptions' package eventually

**Comment by UltronDestroyer** [comment]
I removed the gripper stuff.

**Comment by UltronDestroyer** [comment]
Also the optenv thing for the kinect is now done, so this can be merged.

**Comment by paulbovbel** [comment]
https://github.com/husky/husky_description/pull/11/files#diff-2a507b3d24d51a4dd10ba8613708db2cR31
https://github.com/husky/husky_description/pull/11/files#diff-ba69af255156f3b481175d26f1dace9bR131

**Comment by paulbovbel** [comment]
LGTM, please merge together with husky_simulator and husky_control, once you move the stow script.

**Comment by UltronDestroyer** [comment]
husky_control now has the stow script.

**Pull request merged**